sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightWorkerTask.kt

```diff
@@ -24,5 +24,17 @@ abstract class SqlDelightWorkerTask : SourceTask() {
   internal fun workQueue(): WorkQueue =
     workerExecutor.processIsolation {
       it.classpath.from(classpath)
+
+      it.forkOptions { forkOptions ->
+        // Environment variables are not forwarded to worker processes by default,
+        // see https://github.com/gradle/gradle/issues/8030.
+        // This breaks retrieval of environment variables like temporary directories in the worker
+        // process, which are required for JDBC to work correctly.
+        // Therefore, we explicitly forward all environment variables to the worker process.
+        forkOptions.environment(System.getenv())
```

Comment:
Instead forwarding all environment variables we should add a MapProperty to the task and pass the property here. Then you can opt-in passing your specific environment variables using `tasks.verifyMigrations`/`tasks.withType(SqlDelightWorkerTask::class)`

Reply:
Thank you for your response. I can see that your solution is cleaner and more customizable, yet I have a few concerns.

If we avoided forwarding TEMP environment variables, Any team with developers on Windows machines would have to forward them which adds more code to the minimal setup. Do you think that in addition to adding a property to customize them, we can forward the bare-minimum env variables and increase the maximum heap size by default? Also, the heap size limitation may affect developers on other hosts as well.

Also, to what extent should the tasks be customizable? just `environment` and `maxHeapSize`? Or also add `systemProperties`, `jvmArgs`, and `minHeapSize`? Should we add functions like `environment()`, and `jvmArgs()`?

Reply:
I've made the task more configurable by adding `environment`, `systemProperties`, `minHeapSize`, `maxHeapSize`, and `jvmArgs`. I've made `environment` inherit inheritable env variables from its parent similar to how `DefaultJavaForkOptions` handle it. Also, I've set the default value of `maxHeapSize` to 512MB which handles my average database well, and it is configurable for larger projects.

A further step would be to implement `ClassLoaderWorkerSpec` and `JavaForkOptions` and implement the remaining properties, or `ProcessWorkerSpec` and hide the forkOptions behind an additional property. 

I'd appreciate your review since this affects the API a lot.